src/fetch_business_partners_info.py

```diff
@@ -0,0 +1,104 @@
+import os
+import json
+import datetime
+import time
+import pandas as pd
+from unicodedata import normalize
+from selenium import webdriver
+from bs4 import BeautifulSoup
+
+
+def selenium_driver_path():
+    for path in ['chromedriver.exe', 'chromedriver']:
+        if os.path.exists(path):
+            return path
+    raise BaseException('ChromeDriver not found')
+
+
+def normalize_name(name):
+    return normalize('NFKD', name).encode('ASCII', 'ignore').decode('utf-8').lower()
+
+
+def make_google_cache_url(name):
+    return 'http://webcache.googleusercontent.com/search?btnG=Search&q=cache:www.consultasocio.com/q/sa/%s' % \
+           normalize_name(name).replace(' ', '-')
+
+
+def make_consulta_socio_url(name):
+    return 'http://www.consultasocio.com/q/sa/%s' % normalize_name(name).replace(' ', '-')
+
+
+def extract_info_from_page(url, info, driver, error_msg):
+    # Open the page using Selenium in order to collect the full HTML to be parsed by BeautifulSoup
+    driver.get(url)
+    driver.switch_to.default_content()
+    soup = BeautifulSoup(driver.page_source, 'html.parser')
+
+    # Somethimes google can ask for captcha, so you should type it!
+    if 'To continue, please type the characters below:' in soup.text:
+        while 'To continue, please type the characters below:' in soup.text:
+            driver.switch_to.default_content()
+            soup = BeautifulSoup(driver.page_source, 'html.parser')
+            time.sleep(1)
+
+    # Collect the information if there is no error
+    if error_msg.lower() not in soup.title.text.lower():
+        for partner in soup.findAll('span', attrs={'class': 'socio'}):
+            info['partners'].append(partner.text)
+        for company_info in soup.findAll('p', attrs={'class': 'number'}):
+            info['business'].append(company_info.find('span', {'class': 'text'}).text)
+        try:
+            info['cache_info'] = soup.find('div', attrs={'id': 'google-cache-hdr'}).text
+        except:
+            info['cache_info'] = str(datetime.datetime.now())
```

Comment:
I see that `cache_info` may contain a string (as `'This is Google's cache of http://www.consultasocio.com/q/sa/nilson-aparecido-leitao. It is a snapshot of the page as it appeared on Dec 10, 2016 16:47:53 GMT. The current page could have changed in the meantime. Learn moreFull versionText-only versionView sourceTip: To quickly find your search term on this page, press Ctrl+F or ⌘-F (Mac) and use the find bar.'` or a datetime. If the idea of storing the Google Cache info is for having an option to go back and look for bugs in the future (good practice btw), can you create a new column `collected_at` or something like this, present in all the records, collected or not from the cache?